src/comparator/wrappers.py

```diff
@@ -834,51 +818,86 @@ def __init__(
         # set including the nested messages/enums, since they could also be referenced.
         # Key is the full name of the message/enum and value is the Message/Enum object.
         self._get_global_info_map(source_code_locations_map)
-        # Get all messages in the map.
-        # TODO(xiaozhenliu): update the map when global messages map is in place.
-        self.messages_map = self._get_messages_map(source_code_locations_map)
-
+        # Get all **used** information for comparison.
         self.packaging_options_map = defaultdict(dict)
         self.services_map: Dict[str, Service] = {}
         self.enums_map: Dict[str, Enum] = {}
+        self.messages_map: Dict[str, Message] = {}
         path = ()
-        for fd in file_set_pb.file:
+        for fd in self.definition_files:
             source_code_locations = source_code_locations_map[fd.name]
```

Comment:
Find more explanation in PR description #113 we only iterate the API definition files instead of all proto files. Now the `services_map`, `messages_map`, `enums_map` stand for all the services/messages/enums that are in use.